hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AsyncFSWAL.java

```diff
@@ -323,7 +324,9 @@ private void syncFailed(long epochWhenSync, Throwable error) {
   private void syncCompleted(AsyncWriter writer, long processedTxid, long startTimeNs) {
     highestSyncedTxid.set(processedTxid);
     for (Iterator<FSWALEntry> iter = unackedAppends.iterator(); iter.hasNext();) {
-      if (iter.next().getTxid() <= processedTxid) {
+      FSWALEntry entry = iter.next();
+      if (entry.getTxid() <= processedTxid) {
+        entry.getRpcCall().ifPresent(ServerCall::releaseByWAL);
```

Comment:
I think we can expose an method named  release() in FSWALEntry  and hide the ServerCall#release detail inside it ?   Say, here can just call entry.release(), it may be cleaner ...